testing_and_setup/compass/utility_scripts/LIGHTparticles/make_particle_file.py

```diff
@@ -348,26 +349,79 @@ def write(self, f_name, f_decomp):  #{{{
 #}}}
 
 
-def get_cell_coords(f_init): #{{{
-    return f_init.variables['xCell'][:], \
-           f_init.variables['yCell'][:], \
-           f_init.variables['zCell'][:] #}}}
+def get_particle_coords(f_init, loc=None): #{{{
+    #
+    if loc is None:
+        loc = 'center'
+
+    if loc == 'center':
+        xCell = f_init.variables['xCell'][:]
+        yCell = f_init.variables['yCell'][:]
+        zCell = f_init.variables['zCell'][:]
+        cells = (xCell, yCell, zCell)
+        cpts = np.arange(len(xCell))
+    elif loc == 'offvertex':
+        xVertex = f_init.variables['xVertex'][:]
+        yVertex = f_init.variables['yVertex'][:]
+        zVertex = f_init.variables['zVertex'][:]
+
+        xCell = f_init.variables['xCell'][:]
```

Comment:
Could bump `xCell`, `yCell`, `zCell` outside of these if-statements since they are being defined in both.

Reply:
Agreed